app/controllers/posts_controller.rb

```diff
@@ -7,4 +7,21 @@ def show
     @user = User.find(params[:user_id])
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @author = current_user
+    @post = Post.new(params.require(:post).permit(:title, :text))
```

Comment:
- [OPTIONAL] This is good 👍 But do you know that you can improve on the code by using [strong parameters](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides)

```ruby
private 
def post_params
    params.require(:post).permit(:title, :text)
end
```

That private method can be now pass as the new params
```ruby
@post = @current_user.posts.new(post_params)
```